### PR TITLE
Throw error if more than 1 access token is matched

### PR DESCRIPTION
**Pull request by sangonzal**

### CODE REVIEW

**Review by sameerag — APPROVED** [review]
Lgtm

File: lib/msal-common/src/client/OnBehalfOfClient.ts [hunk 1]
```diff
@@ -88,12 +89,16 @@ export class OnBehalfOfClient extends BaseClient {
...
+        const numAccessTokens = accessTokens.length;
+        if (numAccessTokens < 1) {
             return null;
+        } else if (numAccessTokens > 1) {
+            throw ClientAuthError.createMultipleMatchingTokensInCacheError();
         }
         return accessTokens[0] as AccessTokenEntity;
     }
-    
+
```
Comment (pkanher617) [inline lib/msal-common/src/client/OnBehalfOfClient.ts:101]: Should this be done in the silent flow as well?

**Review by pkanher617 — APPROVED** [review]
Would like to see some tests

File: lib/msal-common/src/client/OnBehalfOfClient.ts (hunk 1 above)
Comment (sangonzal) [reply lib/msal-common/src/client/OnBehalfOfClient.ts:101]: Yes. This logic is no longer in SilentFlowClient, it was refactored into CacheManager. I updated CacheManager as part of this PR.

**Comment by sangonzal** [comment]
Added tests

**Pull request merged**